Simplenote/SimplenoteAppDelegate.m

```diff
@@ -557,4 +572,37 @@ - (NSUndoManager *)windowWillReturnUndoManager:(NSWindow *)window
     return [[self managedObjectContext] undoManager];
 }
 
+
+#pragma mark ================================================================================
+#pragma mark Spotlight
+#pragma mark ================================================================================
+
+- (void)indexSpotlightItemsIfNeeded
+{
+    // This process should be executed *just once*, and only if the user is already logged in (AKA "Upgrade")
+    NSString *kSpotlightDidRunKey = @"SpotlightDidRunKey";
+    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+
+    if ([defaults boolForKey:kSpotlightDidRunKey] == true) {
+        return;
+    }
+
+    [defaults setBool:true forKey:kSpotlightDidRunKey];
+    [defaults synchronize];
+
+    if (self.simperium.user.authenticated == false) {
+        return;
+    }
+
+    [self indexSpotlightItems];
+}
```

Comment:
I'm not 100% sure if we still need this API:

1. In the upgrade flow, I'm seeing the `Spotlight` Option disabled, by default
2. And all of this ends up in `CSSearchableIndex` bailing out

Perhaps we can drop these two API(s)?